libraries/splunk_app.rb

```diff
@@ -126,13 +174,80 @@ def create_splunk_directory(path)
         dir.group(node['splunk']['group'])
         dir.mode('0755')
         dir.run_action(:create)
-        new_resource.updated_by_last_action(dir.updated_by_last_action?)
+        new_resource.updated_by_last_action(true) if dir.updated_by_last_action?
+      end
+
+      def download_and_install # rubocop:disable PerceivedComplexity, CyclomaticComplexity
+        return if new_resource.url.nil? || new_resource.url.empty?
+
+        expected_version = CernerSplunk::AppVersion.new new_resource.version
+        installed_version = CernerSplunk::AppVersion.new @current_resource.version
+
+        return if expected_version.type && expected_version == installed_version
```

Comment:
I don't understand the first part of this expression (`if expected_verison.type`).  `expected_version.type` is not a boolean -- if I read [AppVersion](https://github.com/cerner/cerner_splunk/pull/48/files#diff-940247b1e7418e1207338ac1de247744R36) correctly the 'type' is either `:base` or `:prerelease` and never null?